Prepare for ggplot2 v4

mbstadler opened the pull request.

mbstadler committed:
add labels in plotBinHist conditionally to avoid ggplot2-v4 error

mbstadler committed:
add labels in plotBinDensity conditionally to avoid ggplot2-v4 error

mbstadler committed:
add labels in plotBinScatter conditionally to avoid ggplot2-v4 error

mbstadler committed:
use NULL instead of element_blank() to remove label in labs(...)

mbstadler committed:
add missing imports for element_blank

mbstadler committed:
add unit tests for xlab, ylab and main arguments

mbstadler committed:
use quantiles,quantile.linetype instead of the deprecated draw_quantiles in geom_violin

mbstadler committed:
temprarily install devel version of ggplot2 in GHA to test v4 compatibility

mbstadler committed:
update NEWS.md

mbstadler committed:
add requirement for ggplot2 version (pre-)4

mbstadler committed:
bump version

mbstadler committed:
bump GHA package cache version

mbstadler commented:
I have temporarily added a step to the github actions to install the devel version of `ggplot2`, as well as a version requirement for ggplot2 (>= 3.5.2.9001) to the `DESCRIPTION`. This was mostly required due  to the changes in `geom_violin` (deprecation of `draw_quantiles` in favor of `quantiles` and `quantile.linetype`, etc.), which means that there is no way to implement quantile drawing to work for both pre- and post-v4 ggplot2.

Before we merge (once `ggplot2` 4.0.0 is out), we may want to:

- [x] remove the installation of devel ggplot2 from the github actions
- [x] change the version requirement for ggplot2 in `DESCRIPTION` from >= 3.5.2.9001 to >= 4.0.0

mbstadler committed:
improve plotting unit test coverage

csoneson committed:
Use expect_true(is_ggplot(.)) instead of expect_s3_class(., "ggplot") as recommended in https://github.com/tidyverse/ggplot2/issues/6498

csoneson reviewed (approved):
Looks great to me! I just changed the tests for ggplot objects as recommended in https://github.com/tidyverse/ggplot2/issues/6498 (the other way still works for now, but may be removed in the future)

machlabd commented on tests/testthat/test_plotting.R:
Replace `expect_s3_class` with `expect_true(ggplot2::is_ggplot())` here as well?

https://github.com/fmicompbio/monaLisa/pull/90/files#diff-0a7b5fe1c93e6e3f9b36687b57fd93908dd4f7b8a469a1fee9cafe23cf2f48e1L67-L72

mbstadler replied:
Yes, good catch. Fixed in 5560663c

machlabd commented on R/plotBinDiagnostics.R:
Remove `element_blank` from imports list

https://github.com/fmicompbio/monaLisa/pull/90/files#diff-f18dec138b51d607bce4d640caac55ff5fddf6d3aab635ff2b7f04b60767994dL45-L46

mbstadler replied:
I agree! Done in bd69a298

machlabd reviewed (approved):
Looks great! I've added some minor comments.

mbstadler committed:
replace further expect_s3_class(..., "ggplot") by the recommended expect_true(is_ggplot(...))

mbstadler committed:
remove import for unused element_blank

mbstadler committed:
merge changes from devel (GenomeInfoDb -> Seqinfo)

Merge branch 'devel' into prepare-for-ggplot2-v4

# Conflicts:
#	DESCRIPTION

mbstadler committed:
remove installation of development version of ggplot2

mbstadler committed:
require ggplot2 >= 4.0.0 (argument changes in geom_violin)

mbstadler commented:
As `ggplot2` v4.0.0 is now available, I have removed the installation of ggplot2-devel on GHA and added ggplot2 >= 4.0.0 to `DESCRIPTION`, and will merge and push it upstream as soon as the tests are all green (except for the unrelated issue on bioconductor_docker)

csoneson commented:
Great, thank you! 🎉

machlabd commented:
Perfect, thank you! 🥳
